elixir/apps/web/lib/web/live/policies/components.ex

```diff
@@ -1,9 +1,474 @@
+          <div :if={index > 0} class="text-right mt-3 text-sm text-neutral-900">
+            or
+          </div>
+
+          <div class="col-span-4">
+            <.input
+              type="select"
+              field={condition_form[:values]}
+              name="policy[conditions][remote_ip_location_region][values][]"
+              id={"policy_conditions_remote_ip_location_region_values_#{index}"}
+              options={[{"Select Country", nil}] ++ Domain.Geo.all_country_options!()}
+              value_index={index}
+              value={value}
+            />
+          </div>
+        <% end %>
+      </div>
+    </fieldset>
+    """
+  end
+
+  defp remote_ip_condition_form(assigns) do
+    ~H"""
+    <fieldset class="mb-2">
+      <% condition_form = find_condition_form(@form[:conditions], :remote_ip) %>
+
+      <.input
+        type="hidden"
+        field={condition_form[:property]}
+        name="policy[conditions][remote_ip][property]"
+        id="policy_conditions_remote_ip_property"
+        value="remote_ip"
+      />
+
+      <div
+        class="cursor-pointer"
+        phx-click={
+          JS.toggle_class("hidden",
+            to: "#policy_conditions_remote_ip_condition"
+          )
+          |> JS.toggle_class("hero-chevron-down",
+            to: "#policy_conditions_remote_ip_chevron"
+          )
+          |> JS.toggle_class("hero-chevron-up",
+            to: "#policy_conditions_remote_ip_chevron"
+          )
+        }
+      >
+        <legend>
+          <.icon id="policy_conditions_remote_ip_chevron" name="hero-chevron-down" class="w-5 h-5" />
+          IP address
+        </legend>
+
+        <p class="text-sm text-neutral-500 mb-2">
+          Restrict access based on the client's IP address.
+        </p>
+      </div>
+
+      <div
+        id="policy_conditions_remote_ip_condition"
+        class="grid gap-2 sm:grid-cols-5 sm:gap-4 hidden"
+      >
+        <.input
+          type="select"
+          name="policy[conditions][remote_ip][operator]"
+          id="policy_conditions_remote_ip_operator"
+          field={condition_form[:operator]}
+          placeholder="Operator"
+          options={condition_operator_options(:remote_ip)}
+          value={condition_form && condition_form[:operator].value}
+        />
+
+        <%= for {value, index} <- Enum.with_index((condition_form[:values] && condition_form[:values].value || []) ++ [nil]) do %>
+          <div :if={index > 0} class="text-right mt-3 text-sm text-neutral-900">
+            or
+          </div>
+
+          <div class="col-span-4">
+            <.input
+              type="text"
+              field={condition_form[:values]}
+              name="policy[conditions][remote_ip][values][]"
+              id={"policy_conditions_remote_ip_values_#{index}"}
+              placeholder="189.172.0.0/24"
+              value_index={index}
+              value={value}
+            />
```

Comment:
Any room for field helptext on these?

```suggestion
            />
            <p class="text-xs">
              IP and CIDR ranges accepted.
            </p>
```

Reply:
@jamilbk should we do a placeholder `189.172.0.0/24 or 1050:0:5:600:300c:326bor/24` instead?

Reply:
So I was just thinking about that yesterday -- the problem with that I've seen on calls is it confuses users into thinking that's the actual value that will be used when the field is saved.

Maybe as a rule we shouldn't use actual values for placeholders?

Reply:
Maybe we need to redesign placeholders. It's one of the most common features of HTML forms, if users don't see difference between placeholder and value it's not placeholder issue but the design one

Reply:
Yeah probably a mix of both. We could use `E.g. 189.172.0.0/24` and then it wouldn't be ambiguous. I think Carter had some feedback on placeholders, will see if I can find it